src/wiseagents/wise_agent_impl.py

```diff
@@ -632,6 +636,316 @@ def set_response_delivery(self, response_delivery: Callable[[], WiseAgentMessage
+        """
+        self._response_delivery = response_delivery
+
+
+class CollaboratorWiseAgent(WiseAgent):
+    """
+    This agent implementation is meant to be used in conjunction with a CoordinatorWiseAgent.
+    A collaborator agent will receive a request from a coordinator agent and will process the
+    request, adding its response to the shared context. The collaborator agent will then send
+    the coordinator agent a message to let the coordinator know that it has finished executing
+    its work.
+    """
+    yaml_tag = u'!wiseagents.CollaboratorWiseAgent'
+
+    def __init__(self, name: str, description: str, llm: WiseAgentLLM, transport: WiseAgentTransport):
+        """
+        Initialize the agent.
+
+        Args:
+            name (str): the name of the agent
+            description (str): a description of the agent
+            llm (WiseAgentLLM): the LLM agent to use for processing requests
+            transport (WiseAgentTransport): the transport to use for communication
+        """
+        self._name = name
+        self._description = description
+        self._transport = transport
+        self._llm = llm
+        super().__init__(name=name, description=description, transport=self.transport, llm=llm)
+
+    def __repr__(self):
+        """Return a string representation of the agent."""
+        return f"{self.__class__.__name__}(name={self.name}, description={self.description}, llm={self.llm}, transport={self.transport})"
+
+    def process_event(self, event):
+        """Do nothing"""
+        return True
+
+    def process_error(self, error):
+        """Log the error and return True."""
+        logging.error(error)
+        return True
+
+    def process_request(self, request: WiseAgentMessage):
+        """
+        Process a request message by passing it to the LLM and then send a response back to the sender
+        to let them know the request has been processed.
+
+        Args:
+            request (WiseAgentMessage): the request message to process
+        """
+        ctx = WiseAgentRegistry.get_or_create_context(request.context_name)
+        chat_id = request.chat_id
+        if chat_id is not None:
+            # Get the chat messages so far
+            messages = ctx.llm_chat_completion[chat_id]
+        else:
+            messages = []
+
+        messages.append({"role": "system", "content": self.llm.system_message})
+        messages.append({"role": "user", "content": request.message})
+        llm_response = self.llm.process_chat_completion(messages, [])
+
+        # Add this agent's response to the shared context
+        ctx.append_chat_completion(chat_uuid=chat_id, messages=llm_response.choices[0].message)
+
+        # Let the sender know that this agent has finished processing the request
+        self.send_response(
+            WiseAgentMessage(message=WISE_AGENT_ACK, sender=self.name, context_name=request.context_name,
```

Comment:
Why don't use a specific WiseAgentMessageType and leave the message more free in case ACK need some more details in a specific implementation in the future?

Reply:
Was thinking the same thing earlier. Will update this in a bit.

Reply:
Created https://github.com/wise-agents/wise-agents/issues/216 to track this.